server/chat-commands/core.ts

```diff
@@ -485,13 +485,15 @@ export const commands: ChatCommands = {
 	},
 	noreplyhelp: [`/noreply [command] - Runs the command without displaying the response.`],
 
-	async msgroom(target, room, user, connection) {
+	async msgroom(target, room, user, connection, cmd) {
 		const [targetId, message] = Utils.splitFirst(target, ',').map(i => i.trim());
 		if (!targetId || !message) {
 			return this.parse(`/help msgroom`);
 		}
 		const targetRoom = Rooms.search(toID(targetId));
-		if (!targetRoom) return this.errorReply(`Room not found`);
+		if (!targetRoom) return this.errorReply(`Room not found.`);
+		if (!targetRoom.users[user.id]) return this.errorReply(`You must be in the room "${targetRoom.title}" to message it.`);
```

Comment:
I don't think this is necessary; I think this just makes it harder on Pages. Presumably moderation commands should still work when you're not in a room.

Reply:
fair

Reply:
Main thing i don't like about this is you can currently use it to just send casual messages like 'hi' to rooms you're not in